---
title: "Data Pre-Processing and GWAS"
always_allow_html: yes
fontsize: 12pt
geometry: margin=1in
mainfont: Arial
csl: aha.csl
output: html_notebook
---

# Data Pre-Processing

1. Recode XY pseudoautosomal region with: 

```plink --bfile <file prefix> --make-bed --merge-x --out merge_x```

2. Convert to VCF file

```plink --bfile <file prefix> --recode vcf-iid bgz --out <vcf out prefix> --output-chr MT ```
```tabix <vcffile>```

The last bit tells it to use X, Y, and MY rather than 23, 24, 26

4. Remove 0 chr/pos and apply updated manifest file

clean_vcf.py

Pipe into ```bcftools sort -Oz```

Then index with tabix. 

5. Filter by maf (0.01)

```bcftools view --min-af 0.01 <vcf>```

6. Remove duplicate markers (because these are common in these array files)

```bcftools norm --rm-dup none phs000963.remap.af001.vcf.gz -Oz --threads 4 > phs000963.remap.af001.nodup.vcf.gz```

7. Fix pheno file with correct IDs

8. Run PLINK with MACE phenotype

9. Convert VCF to tped for GenABEL: 

```plink --vcf <vcffile> --make-bed --recode transpose```

(also load into GenABEL object)

```{r eval=FALSE}
convert.snp.tped(tpedfile = "analysis/plink/phs000963.remap.af01.nodup.autosomes.tped", 
                 tfamfile = "analysis/plink/phs000963.remap.af01.nodup.autosomes.tfam", 
                 outfile = "phs000963.remap.af01.nodup.autosomes.genabel", 
                 strand = "+")
```


Prep the ped file with: 

``` awk -F "," 'FNR > 1 {print $1"\t"$1"\t"($4=="F"?2:1)"\t"($10+1)}' phs000963.pheno.csv```

# GWAS Analysis

``` plink --tfile phs000963.remap.af01.nodup.autosomes --pheno MACE_PLINK_PHENO.pheno --make-bed --out phs000963.remap.af01.nodup.autosomes.MACE --allow-no-sex```

```plink --bfile phs000963.remap.af01.nodup.autosomes.MACE --allow-no-sex --assoc```
